Python/Product/PythonTools/PythonTools/Debugger/CustomDebuggerEventHandler.cs

```diff
@@ -83,5 +84,28 @@ public class CustomDebuggerEventHandler : IVsCustomDebuggerEventHandler110 {
                 "interpreter, please disable PGO.";
             MessageBox.Show(content, "PGO Is Not Supported", MessageBoxButton.OK, MessageBoxImage.Error);
         }
+
+        private void SetDebugOptions(IDebugEngine2 engine) {
+            var pyService = _serviceProvider.GetPythonToolsService();
+
+            var options = new StringBuilder();
+            if (pyService.DebuggerOptions.WaitOnAbnormalExit) {
+                options.Append(";" + AD7Engine.WaitOnAbnormalExitSetting + "=True");
+            }
+            if (pyService.DebuggerOptions.WaitOnNormalExit) {
+                options.Append(";" + AD7Engine.WaitOnNormalExitSetting + "=True");
+            }
+            if (pyService.DebuggerOptions.TeeStandardOutput) {
+                options.Append(";" + AD7Engine.RedirectOutputSetting + "=True");
+            }
+            if (pyService.DebuggerOptions.BreakOnSystemExitZero) {
+                options.Append(";" + AD7Engine.BreakSystemExitZero + "=True");
+            }
+            if (pyService.DebuggerOptions.DebugStdLib) {
+                options.Append(";" + AD7Engine.DebugStdLib + "=True");
+            }
+
+            engine.SetMetric(AD7Engine.DebugOptionsMetric, options.ToString());

```

Comment:
On the other hand, this is kinda sorta public API also, because it's on the boundary between Debugger and PythonTools, SetMetric is public, and so is the metric name and option names (exposed as public constants on AD7Engine). Thoughts?